clang/lib/Sema/CheckExprLifetime.cpp

```diff
@@ -1239,11 +1239,12 @@ static AnalysisResult analyzePathForGSLPointer(const IndirectLocalPath &Path,
     }
     // Check the return type, e.g.
     //   const GSLOwner& func(const Foo& foo [[clang::lifetimebound]])
+    //   GSLOwner* func(cosnt Foo& foo [[clang::lifetimebound]])
     //   GSLPointer func(const Foo& foo [[clang::lifetimebound]])
     if (FD &&
-        ((FD->getReturnType()->isReferenceType() &&
+        ((FD->getReturnType()->isPointerOrReferenceType() &&
           isRecordWithAttr<OwnerAttr>(FD->getReturnType()->getPointeeType())) ||
-         isPointerLikeType(FD->getReturnType())))
+          isGLSPointerType(FD->getReturnType())))
```

Comment:
Does it mean that only `gsl::Pointer` return types are supported ?

For example `std::unique_ptr` in `int* x = make_unique<int>(1).get();`
https://godbolt.org/z/73bTWd48b
I think this would stop working ?

Reply:
Since this is on GSL init path, the case mentioned should not be affected.

I also there are related tests already present:
```
std::unique_ptr<int> getUniquePtr();
int *danglingUniquePtrFromTemp() {
  return getUniquePtr().get(); // expected-warning {{returning address of local temporary object}}
}
int *danglingUniquePtrFromTemp2() {
  return std::unique_ptr<int>().get(); // expected-warning {{returning address of local temporary object}}
}
```
So this should be good to go.